web_dwc2.py

```diff
@@ -623,467 +613,251 @@ def rr_reply(self, web_):
-                ]
+                "tools": [],
             })
 
-        if self.get_host_temp() > 0:
-            self.status_3['temps'].update({
-                "extra": self.status_3['temps']['extra'] + [
-                    {
-                        "name": "Host",
-                        "temp": self.get_host_temp()
-                    }
-                ]
+            if self.extruders is not None:
+                index_ = 0
+                for ex_ in self.extruders:
+                    if ex_ is not None:
+                        self.status["tools"].append({
+                            "number": index_,
+                            "name": "Tool" + str(index_),
+                            "heaters": [index_ + 1],  # TODO: Can Klipper has more heaters per tool
+                            "drives": [index_],  # TODO: Can Klipper has more drives per tool
+                            "axisMap": [1],
+                            "fans": 1,
+                            "filament": "",
+                            "offsets": [0, 0, 0]
+                        })
+                        index_ = index_ + 1
+
+            if self.get_host_temp() > 0:
+                self.status.update({
+                    "mcutemp": {
+                        "min": self.host_temp_min,
+                        "cur": self.get_host_temp(),
+                        "max": self.host_temp_max
+                    },
+                })
+
+            # TODO: Power in voltages
+            #"vin": {
+            #    "min": 23.9,
+            #    "cur": 24.1,
+            #    "max": 24.3
+            #}
+        elif (rr_status_type == 3):
+            extrRaw = 0
+            for ex_ in self.extruders:
+                if ex_ is not None:
+                    extrRaw = extrRaw + ex_.extrude_pos
+
+            if extrRaw > self.print_data.get('extr_start', 1):
+                extrRaw = extrRaw - self.print_data.get('extr_start', 1)
+            else:
+                extrRaw = 0
+
+            self.status.update({
+                "currentLayer": self.print_data.get('curr_layer', 1),
+                "currentLayerTime": self.print_data.get('curr_layer_dur', 1),
+                "extrRaw": extrRaw,
+                "fractionPrinted": self.sdcard.get_status(now).get('progress', 0),  # percent done
+                "filePosition": self.sdcard.file_position,
+                "firstLayerDuration":
+                    self.print_data.get('firstlayer_dur', 1) if self.print_data.get('firstlayer_dur', 0) > 0
+                    else self.print_data.get('curr_layer_dur', 1),
+                "firstLayerHeight": self.file_infos.get('running_file', {}).get('firstLayerHeight', 0.2),
+                "printDuration": self.print_data.get('print_dur', 1),
+                "warmUpDuration": self.print_data.get('heat_time', 1),
+                "timesLeft": {
+                    "file": self.print_data['tleft_file'] if 'tleft_file' is self.print_data else "",
```

Comment:
not tested that but looks strange to me.
`if 'tleft_file' is self.print_data else ""`

so my proposal is :
``` self.print_data['tleft_file'] if 'tleft_file' is self.print_data.keys() else 0.01 ```